python/ray/train/_internal/utils.py

```diff
@@ -105,8 +144,7 @@ def construct_train_func(
     Raises:
         ValueError: if the input ``train_func`` is invalid.
     """
-    signature = inspect.signature(train_func)
-    num_params = len(signature.parameters)
+    num_required_params = count_required_parameters(train_func)
```

Comment:
Yep, basically just making sure that you don't pass in a function that requires 2 arguments, since that'll error later after the actor is created. We want to fast fail instead on the driver.